### PR TITLE
Adding asset event as part of the event feature

### PR DESCRIPTION
**Pull request by SiniWilson**
### Requirements

Adding asset event as part of the event feature

### Description of the Change

Creation event sent for the newly creating resources

 ### Benefits

 When user selects dynamodb/kinesis/sqs/s3 as event source, user is able to view the asset associated with the service

### CODE REVIEW

File: core/jazz_assets/config/global-config.json [hunk 1]
```diff
@@ -31,7 +31,11 @@
 		"cloudwatch_event",
 		"swagger_url",
 		"endpoint_url",
-		"dynamodb_stream_arn"
+    "dynamodb_stream_arn",
```
Comment (devsatishm) [inline core/jazz_assets/config/global-config.json]: dynamodb_stream

File: core/jazz_assets/config/global-config.json [hunk 2]
```diff
@@ -31,7 +31,11 @@
...
+    "dynamodb_role_arn",
```
Comment (devsatishm) [inline core/jazz_assets/config/global-config.json]: iam_role

File: core/jazz_assets/config/global-config.json [hunk 3]
```diff
@@ -31,7 +31,11 @@
...
+    "sqs_role_arn",
```
Comment (devsatishm) [inline core/jazz_assets/config/global-config.json]: Not required. It is an iam_role (which we covered already)

File: core/jazz_assets/config/global-config.json [hunk 4]
```diff
@@ -31,7 +31,11 @@
...
+    "sqs_queue_arn",
```
Comment (devsatishm) [inline core/jazz_assets/config/global-config.json]: sqs

File: core/jazz_assets/config/global-config.json [hunk 5]
```diff
@@ -31,7 +31,11 @@
...
+    "kinesis_stream_arn"
```
Comment (devsatishm) [inline core/jazz_assets/config/global-config.json]: kinesis_stream

File: builds/jenkins-build-pack-lambda/Jenkinsfile [hunk 6]
```diff
@@ -288,6 +292,13 @@ node() {
             lambdaEvents.updateLambdaPermissionAndNotification(lambdaARN, s3BucketName, config['event_action_s3'])
           }
 
+          if (config['event_source_dynamodb'] || config['event_source_sqs']) {
+            def event_source = (config['event_source_dynamodb']? 'event_source_dynamodb': 'event_source_sqs' ).split("_")[2]
+            def role_name =  "${configLoader.INSTANCE_PREFIX}-${config['domain']}-${config['service']}-${environment_logical_id}"
```
Comment (devsatishm) [inline builds/jenkins-build-pack-lambda/Jenkinsfile]: Is there a better way to get the ARNs of the roles that get created? `sls --info` may be?

File: core/jazz_assets/config/global-config.json (hunk 1 above)
Comment (SiniWilson) [reply core/jazz_assets/config/global-config.json]: Done

File: core/jazz_assets/config/global-config.json (hunk 2 above)
Comment (SiniWilson) [reply core/jazz_assets/config/global-config.json]: Done

File: core/jazz_assets/config/global-config.json (hunk 3 above)
Comment (SiniWilson) [reply core/jazz_assets/config/global-config.json]: Done

File: core/jazz_assets/config/global-config.json (hunk 4 above)
Comment (SiniWilson) [reply core/jazz_assets/config/global-config.json]: Done

File: core/jazz_assets/config/global-config.json (hunk 5 above)
Comment (SiniWilson) [reply core/jazz_assets/config/global-config.json]: Done

File: builds/jenkins-build-pack-lambda/Jenkinsfile (hunk 6 above)
Comment (SiniWilson) [reply builds/jenkins-build-pack-lambda/Jenkinsfile]: Tried serverless info , list-stack-resouces, get-template etc. Not get the role arn. So going with the current approach. Will keep this and raise a follow up PR, for the same in future.

File: builds/jazz-build-module/aws-lambda-events-module.groovy [hunk 7]
```diff
@@ -58,6 +58,24 @@ def getKinesisStreamArn(stream_name){
...
+      returnStdout: true
+    ).trim()
+    def mappings = parseJson(response)
+    echo "role details : $mappings "
+    if(mappings.Role){
+      role_arn = mappings.Role.Arn
+    }
+    return role_arn
+  } catch (ex) {
+    echo "Error occured while describing the role details"
```
Comment (svidhani) [inline builds/jazz-build-module/aws-lambda-events-module.groovy]: echo ex?

File: builds/jenkins-build-pack-lambda/Jenkinsfile [hunk 8]
```diff
@@ -276,6 +271,15 @@ node() {
               echo "Exception occured while serverless deployment to ${environment_logical_id} environment : $deployOutput"
               error "Exception occured while serverless deployment to ${environment_logical_id} environment"
             }
+          } else {
+            if (config['event_source_s3']) {
+              def s3_bucket_arn = "arn:aws:s3:::${config['event_source_s3']}"
```
Comment (svidhani) [inline builds/jenkins-build-pack-lambda/Jenkinsfile]: region, account?

File: core/jazz_assets/config/global-config.json [hunk 9]
```diff
@@ -31,7 +31,10 @@
 		"cloudwatch_event",
 		"swagger_url",
 		"endpoint_url",
-		"dynamodb_stream_arn"
+    "dynamodb_stream",
+    "iam_role",
+    "sqs",
```
Comment (svidhani) [inline core/jazz_assets/config/global-config.json]: sqs_queue?

File: builds/jenkins-build-pack-lambda/Jenkinsfile (hunk 8 above)
Comment (SiniWilson) [reply builds/jenkins-build-pack-lambda/Jenkinsfile]: s3 arn does not have account and region

File: core/jazz_assets/config/global-config.json (hunk 9 above)
Comment (SiniWilson) [reply core/jazz_assets/config/global-config.json]: As per Satish's comment.

File: builds/jazz-build-module/aws-lambda-events-module.groovy (hunk 7 above)
Comment (SiniWilson) [reply builds/jazz-build-module/aws-lambda-events-module.groovy]: done

File: core/jazz_assets/config/global-config.json (hunk 9 above)
Comment (svidhani) [reply core/jazz_assets/config/global-config.json]: sure but i don t agree with it. look at other resources(eg, dynamodb_stream, iam_role)

Comment (devsatishm) [reply core/jazz_assets/config/global-config.json]: My thought process - 
There aren't any resources under sqs other than queues, while dynamodb, iam etc. have additional resource types (dynamodb tables, dynamodb streams, iam roles, iam users etc.). 

I'm still fine with sqs_queue too.

**Comment by svidhani** [comment]
LGTM

**Comment by rajeevr2715** [comment]
LGTM

**Pull request merged**